chatkit-core/src/test/kotlin/com/pusher/chatkit/rooms/api/JoinedRoomApiTypeMapperTest.kt

```diff
@@ -0,0 +1,77 @@
+package com.pusher.chatkit.rooms.api
+
+import assertk.assertThat
+import assertk.assertions.isEqualTo
+import assertk.assertions.isNull
+import org.spekframework.spek2.Spek
+import org.spekframework.spek2.style.specification.describe
+
+class JoinedRoomApiTypeMapperTest : Spek({
+
+    describe("given a complete JoinedRoomApiType object") {
+        val joinedRoomApiType = JoinedRoomApiType("1", "ham",
+                "mycoolroom", "pushNotificationOverride", false,
+                mapOf("background-colour" to "red"), "2017-04-14T14:00:42Z",
+                "2017-04-13T14:10:38Z", "2017-04-13T14:10:38Z", null)
+
+        describe("when parsed as a JoinedRoomInternalType object") {
+            val joinedRoomInternalType = JoinedRoomApiTypeMapper().toRoomInternalType(joinedRoomApiType)
+
+            it("then the id will match") {
+                assertThat(joinedRoomInternalType.id).isEqualTo(joinedRoomApiType.id)
+            }
+
+            it("then the name will match") {
+                assertThat(joinedRoomInternalType.name).isEqualTo(joinedRoomApiType.name)
+            }
+
+            it("then the privacy will match") {
```

Comment:
Hmm. I think I preferred individual assertions here so it'd flag more easily what was broken. It was useful for me because it identified the timestamp part was broken more easily when it ran on CI and that saved me a bunch of time :) I've also got a second set of tests to determine that null is expected for some optional values. 

Have updated the `it` descriptions to be more terse.